cpp/src/arrow/memory_pool.cc

```diff
@@ -389,6 +398,30 @@ MemoryPool* default_memory_pool() {
 #endif
 }
 
+#define RETURN_IF_JEMALLOC_ERROR(ERR)                  \
+  do {                                                 \
+    if (err != 0) {                                    \
+      return Status::UnknownError(std::strerror(ERR)); \
+    }                                                  \
+  } while (0)
+
+Status jemalloc_set_decay_ms(int ms) {
```

Comment:
Can we add a test for this? The `mallctl` calls are not trivial.

Reply:
done